src/Database/Connection.php

```diff
@@ -348,7 +392,7 @@ public function insert(string $table, array $values, array $types = []): Stateme
      */
     public function update(string $table, array $values, array $conditions = [], array $types = []): StatementInterface
     {
-        return $this->newQuery()->update($table)
+        return $this->newUpdateQuery()->update($table)
```

Comment:
Should this be :shrug: 
```suggestion
        return $this->newUpdateQuery($table)
```

Seems like we'll be giving a more useful API that is still type safe.